src/shared/components/palette/complex/ccd-read-fields-filter.pipe.ts

```diff
@@ -59,6 +59,10 @@ export class ReadFieldsFilterPipe implements PipeTransform {
     if (field.field_type.type === 'Label' && (field.label || '').length > 0) {
       return true;
     }
+    // We also shouldn't ditch CasePaymentHistoryViewer fields.
+    if (field.field_type.type === 'CasePaymentHistoryViewer') {
+      return true;
+    }
```

Comment:
I wonder whether the "keep fields" could be put into an array to keep them together etc.

const keepFields = ['Label', 'CasePaymentHistoryViewer'] etc...

(just a suggestion) to reduce the if statements.

Reply:
The `'Label'` one has an additional check, or else that's precisely what I would have done. 🤷‍♂️